railties/lib/rails/generators/app_base.rb

```diff
@@ -633,7 +633,11 @@ def target_rails_prerelease(self_command = "new")
 
           run_bundle
 
-          @argv[0] = destination_root
+          if @argv[0].start_with?("-") # rails new --some --flags APP_PATH
+            @argv.insert(0, destination_root)
```

Comment:
That would be ideal, but I wasn't sure how to do that. This works, but maybe isn't the best solution.

This works because by prepending the path, any other destination arguments (APP_PATH in the comment above) are ignored.